### ISSUE TITLE
Allow overriding diagnostic rule settings for a specific directory

### ISSUE THREAD

**Issue by stinodego** [microsoft/pyright#6815]
I'd like to disable `reportUnusedExpression` for our testing code only, as we often have expressions that do not need to be 'used'. An example:

```python
def test_my_function():
    with pytest.raises(ValueError):
        my_function(illegal_input)
```

However, it seems I cannot disable diagnostics for certain parts of the code only. Am I missing something? Otherwise, please consider this a feature request!

**Comment by erictraut** [comment]
You can disable individual diagnostic rules on a per-file basis by adding a `# pyright: reportUnusedExpression=false` comment somewhere in the file (typically near the top).

There's currently no way to override diagnostic rule settings for a specific directory. The configuration for diagnostics in pyright is already extremely complex because we need to compose configuration options from many sources, so it's unlikely that we'd add additional complexity to enable per-directory overrides.
